lib/actions/map.js

```diff
@@ -47,11 +46,26 @@ export function clearLocation(payload) {
   }
 }
 
-export function zoomToStop(stop) {
-  return function (dispatch, getState) {
-    if (!stop) return
-    dispatch(setMapZoom({ zoom: 17 }))
-    dispatch(setMapCenter({ lat: stop.lat, lon: stop.lon }))
+/**
+ * Centers the given map to the coordinates of the specified place.
+ */
+export function setMapCenter(map /* MapRef */, location, zoom) {
+  return function () {
+    const { lat, lon } = location
+    if (map && !isNaN(lat) && !isNaN(lon)) {
+      map.jumpTo({ center: [lon, lat], zoom })
+    }
+  }
+}
+
+/**
+ * Animates the map to the specified stop and the specified zoom level.
+ */
+export function zoomToStop(map /* MapRef */, stop, zoom) {
+  return function () {
+    if (stop && map) {
+      map.flyTo({ center: [stop.lon, stop.lat], zoom: zoom || 17 })
+    }
```

Comment:
No. This is a temporary hack until we can move all zooming and panning into otp-ui. This will happen once the stop viewer is replaced by the nearby page